torch/_inductor/fx_passes/serialized_patterns/central_index.py

```diff
@@ -1,123 +0,0 @@
-from ._sfdp_pattern_17 import (_sfdp_pattern_17_training, _sfdp_pattern_17_inference, _sfdp_pattern_17_half_training, _sfdp_pattern_17_half_inference)
-from ._sfdp_pattern_18 import (_sfdp_pattern_18_training, _sfdp_pattern_18_inference, _sfdp_pattern_18_bs1_training, _sfdp_pattern_18_bs1_inference, _sfdp_pattern_18_half_training, _sfdp_pattern_18_half_inference, _sfdp_pattern_18_half_bs1_training, _sfdp_pattern_18_half_bs1_inference)
-
-central_index = {
-    '_sfdp_pattern_1_training': _sfdp_pattern_1_training,
-    '_sfdp_pattern_1_inference': _sfdp_pattern_1_inference,
-    '_sfdp_pattern_2_training': _sfdp_pattern_2_training,
-    '_sfdp_pattern_2_inference': _sfdp_pattern_2_inference,
-    '_sfdp_pattern_3_training': _sfdp_pattern_3_training,
-    '_sfdp_pattern_3_inference': _sfdp_pattern_3_inference,
-    '_sfdp_pattern_4_training': _sfdp_pattern_4_training,
-    '_sfdp_pattern_4_inference': _sfdp_pattern_4_inference,
-    '_sfdp_pattern_5_training': _sfdp_pattern_5_training,
-    '_sfdp_pattern_5_inference': _sfdp_pattern_5_inference,
-    '_sfdp_pattern_6_training': _sfdp_pattern_6_training,
-    '_sfdp_pattern_6_inference': _sfdp_pattern_6_inference,
-    '_sfdp_pattern_7_training': _sfdp_pattern_7_training,
-    '_sfdp_pattern_7_inference': _sfdp_pattern_7_inference,
-    '_sfdp_pattern_8_training': _sfdp_pattern_8_training,
-    '_sfdp_pattern_8_inference': _sfdp_pattern_8_inference,
-    '_sfdp_pattern_9_training': _sfdp_pattern_9_training,
-    '_sfdp_pattern_9_inference': _sfdp_pattern_9_inference,
-    '_sfdp_pattern_10_training': _sfdp_pattern_10_training,
-    '_sfdp_pattern_10_inference': _sfdp_pattern_10_inference,
-    '_sfdp_pattern_11_training': _sfdp_pattern_11_training,
-    '_sfdp_pattern_11_inference': _sfdp_pattern_11_inference,
-    '_sfdp_pattern_12_training': _sfdp_pattern_12_training,
-    '_sfdp_pattern_12_inference': _sfdp_pattern_12_inference,
-    '_sfdp_pattern_13_training': _sfdp_pattern_13_training,
-    '_sfdp_pattern_13_inference': _sfdp_pattern_13_inference,
-    '_sfdp_pattern_14_training': _sfdp_pattern_14_training,
-    '_sfdp_pattern_14_inference': _sfdp_pattern_14_inference,
-    '_sfdp_pattern_15_training': _sfdp_pattern_15_training,
-    '_sfdp_pattern_15_inference': _sfdp_pattern_15_inference,
-    '_sfdp_pattern_16_training': _sfdp_pattern_16_training,
-    '_sfdp_pattern_16_inference': _sfdp_pattern_16_inference,
-    '_sfdp_pattern_16_bs1_training': _sfdp_pattern_16_bs1_training,
-    '_sfdp_pattern_16_bs1_inference': _sfdp_pattern_16_bs1_inference,
-    '_sfdp_pattern_17_training': _sfdp_pattern_17_training,
-    '_sfdp_pattern_17_inference': _sfdp_pattern_17_inference,
-    '_sfdp_pattern_18_training': _sfdp_pattern_18_training,
-    '_sfdp_pattern_18_inference': _sfdp_pattern_18_inference,
-    '_sfdp_pattern_18_bs1_training': _sfdp_pattern_18_bs1_training,
```

Comment:
They should be - there's just no central index any more (which is what I thought you asked for).  As long as they're listed by _get_sfdp_patterns() (which was needed for them to be generated previously) then they should still be generated.

Reply:
Right no central index is great. We still want a way of updating the patterns. The num_users were not changed, so I'm suspicious we are not generating them as part of this script anymore

Reply:
I'm not sure why `num_users` would change (OTOH - I also don't actually know what `num_users` tracks) because the only thing that changed was how we generate the scripts not any usage of those scripts. But the patterns are definitely being updated. If I manually change it and then regen then they're being re-written:
```
(base) $ hg diff
diff --git a/torch/_inductor/fx_passes/serialized_patterns/_sfdp_pattern_18.py b/torch/_inductor/fx_passes/serialized_patterns$
--- a/torch/_inductor/fx_passes/serialized_patterns/_sfdp_pattern_18.py
+++ b/torch/_inductor/fx_passes/serialized_patterns/_sfdp_pattern_18.py
@@ -205,6 +205,7 @@
 view_default_11 = CallFunction(aten.view.default, bmm_default_5, Ignored())
 permute_default_11 = CallFunction(aten.permute.default, view_default_11, Ignored())
 _sfdp_pattern_18_bs1_training = MultiOutputPattern([view_default_5,
+  "MANUALLY EDITED JUNK",
   permute_default_1,
   permute_default_3,
   permute_default_6,
(base) $ python torchgen/fuse_attention_patterns/gen_attention_patterns.py
(base) $ hg diff
(base) $
```